src/ray/gcs/gcs_server/gcs_resource_manager.cc

```diff
@@ -386,30 +386,34 @@ void GcsResourceManager::GetResourceUsageBatchForBroadcast(
     rpc::ResourceUsageBroadcastData &buffer) {
   absl::MutexLock guard(&resource_buffer_mutex_);
   resources_buffer_proto_.Swap(&buffer);
-  if (!resources_buffer_.empty()) {
-    for (auto &resources : resources_buffer_) {
-      buffer.add_batch()->mutable_data()->Swap(&resources.second);
-    }
-    resources_buffer_.clear();
+  while (!resources_buffer_.empty() &&
+         buffer.ByteSizeLong() <
+             RayConfig::instance().resource_broadcast_batch_size_bytes()) {
+    auto element = std::begin(resources_buffer_);
+    buffer.add_batch()->mutable_data()->Swap(&element->second);
+    resources_buffer_.erase(element);
```

Comment:
Can we reuse GetResourceUsageBatchForBroadcast_Locked here?

Reply:
It's not easy to do. They have different buffer type, one is `ResourceUsageBatchData`, another is `ResourceUsageBroadcastData`